### ISSUE TITLE
Import and Annotate on PDFs directly

### ISSUE THREAD

**Issue by m0byn** [adil192/saber#429]
It would be amazing if it was possible to import a PDF into Saber and annotate directly on it. In more detail, importing not into a note with the PDF being displayed on a page, but importing the PDF without showing it on a page and annotate on it.

**Comment by WasteOfO2** [comment]
That functionality already exists, unless you are talking about saving the annotations made by saving it on the PDF, or by exporting it to a PDF

**Comment by m0byn** [comment]
As far as I can figure out it is only possible to import a PDF into a note and show the PDF pages on a note. However, I am talking about not importing it into a note. So, showing the PDF without a "page" background and editing the PDF.

**Comment by diego-mrqs** [comment]
Maybe related to #277

**Comment by WasteOfO2** [comment]
Is it related to the grid lines showing in ur PDF similar to https://github.com/adil192/saber/issues/400 ?

**Comment by m0byn** [comment]
I do not mean editing PDFs per se like in #277. Rather I am talking about not importing the PDF and scaling it to a note, but importing the PDF, not scaling it to a note, and annotating on it. I am aware, that not setting it as a background image does not scale it. Then, however, it is small and shown on a page of a note. Is it understandable what I am talking about?

**Comment by WasteOfO2** [comment]
I am not 100% sure what u are saying.

Are u talking about editing PDFs like u do it in Adobe Acrobat?

**Comment by m0byn** [comment]
No, let me try to explain in detail. So, when importing a PDF (e.g. of size A4) to saber it automatically sets it as a background, and also scales it to fit the page. As a result, the resolution becomes less clear. The other option is not selecting the PDF as a background. Then, the resolution is clear, but the PDF is smaller than the page. If I open the PDF in e.g. "Secure PDF Viewer" from GrapheneOS, then the PDF shows high resolution and fits on one page. Now, take a PDF of different size (not A4, but any other size), then the scaling also happens. If it is not scaled, then it is shown on a page of the note. What I want is the PDF not scaled in any way such that the resolution stays the original, and also not showing it on a page (just show the PDF, not on a page of a note), and be able to edit it. 

Is it clearer now, what I mean? I could try to provide some screenshots as well!

**Comment by WasteOfO2** [comment]
So ur main concern is the resolution?

**Comment by diego-mrqs** [comment]
So, does adding another page size solve your problem?

**Comment by m0byn** [comment]
My main concern is the resolution when the pdf is scaled to fit on the page, and showing the white page when the pdf is not scaled to fit on the page. I am going to post screenshots now.

**Comment by m0byn** [comment]
This is how it looks like when setting as background. 
![Screenshot_20230301-133519](https://user-images.githubusercontent.com/33026003/222141961-e920de77-1627-4168-94ea-e03460dc81b5.jpg)

The resolution is not that good. 
This is how it looks like when not setting as background.
![Screenshot_20230301-133719](https://user-images.githubusercontent.com/33026003/222142142-0df918e1-818d-40eb-8b4f-3535fec967af.jpg)

It shows the rest of the page of the note.

This is how it looks like in Secure PDF Viewer.
![Screenshot_20230301-133735_PDF Viewer](https://user-images.githubusercontent.com/33026003/222142322-7e8343d5-007f-43fa-8f99-f3f8b3ef1a35.jpg)

Scaled and good resolution and no rest of page showing.

The reason why I opened the issue is, that I have to annotate many PDF files and send them back to the people I got them from. I do not want to send them back in a worse resolution or showing the rest of the pages of the note...

**Comment by WasteOfO2** [comment]
So to understand your use case, you want to:

1. Create a note
2. Import a PDF with good resolution
3. Make annotations on your PDF
4. Update the PDF with annotations with good resolution
5. Send the updated PDF to whoever you want

Am I getting this right?

**Comment by WasteOfO2** [comment]
> Actuallry, you could even skip point 1. So,
> 
>     1. Import the original PDF with good resolution
> 
>     2. Make annotations on your PDF
> 
>     3. Send the imported PDF to whoever you want

Well to make it easier to implement and intuitive for ppl, u do need step 1

**Comment by m0byn** [comment]
Yes, this is correct.

**Comment by WasteOfO2** [comment]
Sure, I can later open separate issues for whatever is mentioned above.

You can close this issue for now, as one of ur issue mentioned is a known issue, while the other is an enhancement.

Yes, I think we can think of a way to dynamically set page sizes according to the PDF

**Comment by m0byn** [comment]
Thank you very much for your help! Just to let you know I am enjoying Saber very much!
